### PR TITLE
Add ARG ARCH in Dockerfile of bootstrap-legacy and golang-legacy

### PR DESCRIPTION
**Pull request by zhlhahaha**
Amend Dockerfile of bootstrap-legacy and golang-legacy to enable multiarch image build

### CODE REVIEW

**Comment by zhlhahaha** [comment]
cc: @xpivarc @dhiller @brianmcarey

File: images/bootstrap-legacy/Dockerfile [hunk 1]
```diff
@@ -94,9 +94,10 @@ RUN mkdir /docker-graph
 #
 
 # Cache the most commonly used bazel versions in the container
-RUN  curl -Lo ./bazelisk https://github.com/bazelbuild/bazelisk/releases/download/v1.7.4/bazelisk-linux-amd64 && \
-     chmod +x ./bazelisk && mv ./bazelisk /usr/local/bin/bazelisk && \
-     cd /usr/local/bin && ln -s bazelisk bazel
+ARG ARCH
```
Comment (dhiller) [inline images/bootstrap-legacy/Dockerfile:97]: nit: I think a default would make sense.
```suggestion
ARG ARCH=amd64
```

File: images/golang-legacy/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,5 @@
-FROM bootstrap-legacy
+ARG ARCH
```
Comment (dhiller) [inline images/golang-legacy/Dockerfile:1]: nit: I think a default would make sense.
```suggestion
ARG ARCH=amd64
```

File: images/bootstrap-legacy/Dockerfile [hunk 3]
```diff
@@ -94,9 +94,10 @@ RUN mkdir /docker-graph
...
+RUN curl -Lo ./bazelisk https://github.com/bazelbuild/bazelisk/releases/download/v1.7.4/bazelisk-linux-${ARCH} && \
```
Comment (dhiller) [inline images/bootstrap-legacy/Dockerfile:98]: nit: bazelisk for bootstrap is version `v1.17.0`, we should use the same on the legacy image
```suggestion
RUN curl -Lo ./bazelisk https://github.com/bazelbuild/bazelisk/releases/download/v1.17.0/bazelisk-linux-${ARCH} && \
```

**Review by dhiller — APPROVED** [review]
Only nits found, decide whether you want to address them. Unhold in the latter case.

/approve
/hold

**Comment by zhlhahaha** [comment]
Thanks Daniel, I have addressed them. The tests `build-bootstrap-legacy-images` will pass once we get the PR https://github.com/kubevirt/project-infra/pull/2952 merged.
/unhold

**Comment by zhlhahaha** [comment]
@dhiller we may need to override the `build-bootstrap-legacy-images` to get this PR merged.

**Review by dhiller — APPROVED** [review]
/override build-bootstrap-legacy-images

**Comment by dhiller** [comment]
@zhlhahaha I guess that since we are pushing new images to quay, the repos for them are implicitly created as private, so we might suffer from pull failures at first.

**Comment by dhiller** [comment]
I'll try to watch these and make them public as soon as they are created

**Comment by zhlhahaha** [comment]
> I'll try to watch these and make them public as soon as they are created

Great thanks, Daniel.

**Pull request merged**